aea/decision_maker/base.py

```diff
@@ -429,35 +460,126 @@ def _split_tx_fees(tx_fee: int) -> Dict[str, int]:
         seller_part = math.ceil(tx_fee * (1 - SENDER_TX_SHARE))
         if buyer_part + seller_part > tx_fee:
             seller_part -= 1
-        return {"seller_tx_fee": seller_part, "buyer_tx_fee": buyer_part}
+        tx_fee_split = {"seller_tx_fee": seller_part, "buyer_tx_fee": buyer_part}
+        return tx_fee_split
+
+
+class ProtectedQueue(Queue):
```

Comment:
Added a protected queue so skills cannot mess with the internal messages submitted by other skills